### PR TITLE
speed improvements 

### PR DESCRIPTION
**Pull request by Bishoy-at-pieces**

### CODE REVIEW

File: src/pieces/api/api_functions.py [hunk 1]
```diff
@@ -33,8 +33,9 @@ def get_models_ids() -> Dict[str, Dict[str, Union[str, int]]]:
     
 
     
+    
     # call the api until the sdks updated
-    response = urllib.request.urlopen('http://localhost:1000/models').read()
+    response = urllib.request.urlopen(f'{host}/models').read()
```
Comment (hal-8999-alpha) [inline src/pieces/api/api_functions.py:38]: Nice

**Review by hal-8999-alpha — COMMENTED** [review]
Looks good to me. Mostly looks like just a good refactor. Where are you seeing the speed improvements? Anything majorly noticeable?

**Comment by Bishoy-at-pieces** [comment]
so basically it take around 5 sec in the first 5 lines which are the import statements I don't think that it is a good thing

**Review by hal-8999-alpha — COMMENTED** [review]
I know Caleb left some comments regarding the import statements. Lmk if you want me to merge this or wait until some of those things have been added

**Comment by Bishoy-at-pieces** [comment]
Now wait

**Pull request merged**